network.go

```diff
@@ -521,6 +527,13 @@ func NetworkOptionDeferIPv6Alloc(enable bool) NetworkOption {
 	}
 }
 
+// NetworkOptionDefaultGW returns an option setter to set if a network needs default gateway
+func NetworkOptionDefaultGW(defaultGW bool) NetworkOption {
```

Comment:
Can we call it something like `NetworkOptionIsolatedNetwork`

As a network property, we do not need to talk about default gw network, that is an implementation detail.
I think I read it somewhere people asking for this feature to refer to it as `isolated` network.